lib/index.js

```diff
@@ -47,218 +45,202 @@ const loginWeb = (opener, opts = {}) => {
 const isValidUrl = u => {
   try {
     return /^https?:$/.test(new URL(u).protocol)
-  } catch (er) {
+  } catch {
     return false
   }
 }
 
-const webAuth = (opener, opts, body) => {
-  const { hostname } = opts
-  body.hostname = hostname || os.hostname()
-  const target = '/-/v1/login'
-  const doneEmitter = new EventEmitter()
-  return fetch(target, {
-    ...opts,
-    method: 'POST',
-    body,
-  }).then(res => {
-    return Promise.all([res, res.json()])
-  }).then(([res, content]) => {
-    const { doneUrl, loginUrl } = content
+const webAuth = async (opener, opts, body) => {
+  try {
+    const res = await fetch('/-/v1/login', {
+      ...opts,
+      method: 'POST',
+      body: {
+        ...body,
+        hostname: opts.hostname || os.hostname(),
+      },
+    })
+
+    const content = await res.json()
     log.verbose('web auth', 'got response', content)
+
+    const { doneUrl, loginUrl } = content
     if (!isValidUrl(doneUrl) || !isValidUrl(loginUrl)) {
       throw new WebLoginInvalidResponse('POST', res, content)
     }
-    return content
-  }).then(({ doneUrl, loginUrl }) => {
-    log.verbose('web auth', 'opening url pair')
-
-    const openPromise = opener(loginUrl, doneEmitter)
-    const webAuthCheckPromise = webAuthCheckLogin(doneUrl, { ...opts, cache: false })
-      .then(authResult => {
-        log.verbose('web auth', 'done-check finished')
-
-        // cancel open prompt if it's present
-        doneEmitter.emit('abort')
-
-        return authResult
-      })
-
-    return Promise.all([openPromise, webAuthCheckPromise]).then(
-      // pick the auth result and pass it along
-      ([, authResult]) => authResult
-    )
-  }).catch(er => {
-    // cancel open prompt if it's present
-    doneEmitter.emit('abort')
 
+    return await webAuthOpener(opener, loginUrl, doneUrl, opts)
+  } catch (er) {
     if ((er.statusCode >= 400 && er.statusCode <= 499) || er.statusCode === 500) {
       throw new WebLoginNotSupported('POST', {
         status: er.statusCode,
         headers: { raw: () => er.headers },
       }, er.body)
-    } else {
-      throw er
     }
-  })
+    throw er
+  }
+}
+
+const webAuthOpener = async (opener, loginUrl, doneUrl, opts) => {
+  const abortController = new AbortController()
+  try {
+    log.verbose('web auth', 'opening url pair')
+    const [, authResult] = await Promise.all([
```

Comment:
This feels clunky, Is there a compelling reason we're racing these two commands together?  check login can't do anything till the opener is done anyways.

Reply:
The only benefit of awaiting them in tandem is that a ctrl-c from the user aborts the opener attempt?